src/main.rs

```diff
@@ -277,17 +279,61 @@ fn isnt_empty(&(_, language): &(&LanguageType, &Language)) -> bool {
 fn print_language<W>(sink: &mut W,
                      lang_section_len: usize,
                      language: &Language,
-                     name: &str)
+                     total: &Language,
+                     name: &str,
+                     show_percentage: bool)
     -> io::Result<()>
     where W: Write,
 {
-    writeln!(sink,
-             " {:<len$} {:>6} {:>12} {:>12} {:>12} {:>12}",
-             name,
-             language.stats.len(),
-             language.lines,
-             language.code,
-             language.comments,
-             language.blanks,
-             len = lang_section_len)
+    if !show_percentage {
+       return writeln!(
+                 sink,
+                 " {:<len$} {:>6} {:>12} {:>12} {:>12} {:>12}",
+                 name,
+                 language.stats.len(),
+                 language.lines,
+                 language.code,
+                 language.comments,
+                 language.blanks,
+                 len = lang_section_len)
+    }
+
+    let percent_char_count = 5;
+    // don't show percentage for the total row because that would be pointless
+    if language.lines == total.lines {
```

Comment:
Instead of passing in total and performing this if statement, why not just call ` print_language(&mut stdout, columns - NO_LANG_ROW_LEN, &total, &total, "Total", percent_option)?;` as ` print_language(&mut stdout, columns - NO_LANG_ROW_LEN, &total, &total, "Total", false)?;`?

Reply:
I think then that there should probably be a seperate function for printing total. Currently this implementation changes a lot of behaviour (such as moving the total around which adds additional allocations in `AddAssign`) which is not desired.

Reply:
How would you suggest I do that?

Reply:
So I've refactored all the printing code into `src/cli_utils.rs` so you'll need to move your changes to that. I would suggest having a function called `print_total` that handles all the logic of printing out the total.

Reply:
I'm not able to rework this now, so I'll pick it up again later when I can. I'll just go ahead and close the PR until then. thanks

Reply:
@BrainMaestro Okay, thank you again for your PR! I'm going to take sometime and write a comment on the issue detailing exact requirements so that if and when you do get back to it you'll have a much clearer sense of the task.

Reply:
@Aaronepower that sounds great.

Reply:
> I'm going to take some time and write a comment on the issue detailing exact requirements

Hi, was this ever written? I'm interested in trying to (re)implement this. Although I'm new to Rust and open source.